lib/src/aggregate_sample.dart

```diff
@@ -5,41 +5,54 @@
 import 'dart:async';
 
 extension AggregateSample<T> on Stream<T> {
```

Comment:
Is this method not exported by the `async.dart` library?

If it's internal, then most of the comments here can be ignored (like all constructor arguments being required is fine, I'd personally make them positional.)